modules/rubiconBidAdapter.js

```diff
@@ -885,21 +885,22 @@ function addVideoParameters(data, bidRequest) {
 function applyFPD(fpd, mediaType, data) {
   const map = {user: {banner: 'tg_v.', code: 'user'}, context: {banner: 'tg_i.', code: 'site'}, adserver: 'dfp_ad_unit_code'};
   let obj = {};
+  let impData = {};
   let keywords = [];
-  const validate = function(e, t) {
-    if (typeof e === 'object' && !Array.isArray(e)) {
-      utils.logWarn('Rubicon: Filtered FPD key: ', t, ': Expected value to be string, integer, or an array of strings/ints');
-    } else if (e) {
-      return (Array.isArray(e)) ? e.filter(value => {
-        if (typeof value !== 'object' && value) return value;
-
-        utils.logWarn('Rubicon: Filtered value: ', value, 'for key', t, ': Expected value to be string, integer, or an array of strings/ints');
-      }).toString() : e.toString();
+  const validate = function(prop, key) {
+    if (typeof prop === 'object' && !Array.isArray(prop)) {
+      utils.logWarn('Rubicon: Filtered FPD key: ', key, ': Expected value to be string, integer, or an array of strings/ints');
+    } else if (utils.isNumber(prop) || prop) {
```

Comment:
I think a simple 

```
prop || prop === 0
```

would be a little more efficient. Most of the time the prop will not be falsey, so we should check that first regardless.

And we know it will only hit the other side of the `||` if it is falsey, and only falsey number is zero so better to strict check that.

Reply:
I believe this is outdated as well